src/csharp/Microsoft.Spark.Worker/SimpleWorker.cs

```diff
@@ -22,15 +22,20 @@ internal SimpleWorker(Version version)
         }
 
         internal void Run()
+        {
+            int port = Utils.SettingUtils.GetWorkerFactoryPort(_version);
+            Run(port);
+        }
+
+        internal void Run(int port)
         {
             try
             {
-                int port = Utils.SettingUtils.GetWorkerFactoryPort(_version);
                 string secret = Utils.SettingUtils.GetWorkerFactorySecret(_version);
+                ISocketWrapper socket = SocketFactory.CreateSocket();
```

Comment:
can you move this back above `socket.Connect(IPAddress.Loopback, port, secret);` ?

Reply:
resolved at 89cc504